yt_dlp/extractor/arte.py

```diff
@@ -217,43 +216,39 @@ class ArteTVPlaylistIE(ArteTVBaseIE):
     _VALID_URL = r'https?://(?:www\.)?arte\.tv/(?P<lang>%s)/videos/(?P<id>RC-\d{6})' % ArteTVBaseIE._ARTE_LANGUAGES
     _TESTS = [{
         'url': 'https://www.arte.tv/en/videos/RC-016954/earn-a-living/',
-        'info_dict': {
-            'id': 'RC-016954',
-            'title': 'Earn a Living',
-            'description': 'md5:d322c55011514b3a7241f7fb80d494c2',
-        },
-        'playlist_mincount': 6,
+        'only_matching': True,
     }, {
         'url': 'https://www.arte.tv/pl/videos/RC-014123/arte-reportage/',
+        # too long to comfortably test
         'only_matching': True,
     }]
 
     def _real_extract(self, url):
         lang, playlist_id = self._match_valid_url(url).groups()
-        collection = self._download_json(
-            '%s/collectionData/%s/%s?source=videos'
+        playlist = self._download_json(
+            '%s/playlist/%s/%s'
             % (self._API_BASE, lang, playlist_id), playlist_id)
+        metadata = playlist['data']['attributes']['metadata']
         entries = []
-        for video in collection['videos']:
+        for video in playlist['data']['attributes']['items']:
             if not isinstance(video, dict):
                 continue
-            video_url = url_or_none(video.get('url')) or url_or_none(video.get('jsonUrl'))
+            video_url = video['config']['url']
             if not video_url:
                 continue
```

Comment:
I am kind of reluctant to `traverse_obj` and the like, since I am never sure what those helpers actually do when I encounter them. This is just too much magic for one function.

Reply:
`video_url = try_get(video, lambda x: x['config']['url'])` or its `traverse_obj()` equivalent has the possible advantage of not crashing the extraction if one item should have an _unexpected_ structure.

Reply:
You have misspelt ‘disadvantage’. Masking errors, especially when extracting the most crucial datum, the video URL, is a bug, pure and simple, that should be surfaced as early as possible, reported and eventually fixed. Even PHP is moving away from the PHP school of error handling (pretend nothing happened and hope for the best).

Reply:
"possible" vs "dis-"
If the structure were controlled by us, I agree. But one novel item structure shouldn't break whatever extraction still works: maybe a WARNING, though. If no video URLs are found at all then a crash is quite reasonable and the `_sort_formats()` method provides it.